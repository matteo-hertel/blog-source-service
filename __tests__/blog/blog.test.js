require('dotenv').config({
    path: `${__dirname}/../../.env`
});
jasmine.DEFAULT_TIMEOUT_INTERVAL = 15000;
const blog = require(`${__dirname}/../../modules/blog`);

describe("Blog Module", () => {
    it('should return a storable payload for a given file', () => {
        return blog.getStorable("matteo-hertel", "blog", "master", "test/posts/2017-01-27_test-post-seo.md")
        .then((data) => {
            data.updatedDate =  1482363367071;
            expect(data).toMatchSnapshot("storableFile");
        });
    });
    it('should execute the catch if the file does not exists', () => {
        return blog.getStorable("does/not/exists")
        .catch((exc) => {
            expect(exc).toBeDefined();
        });
    });
});
