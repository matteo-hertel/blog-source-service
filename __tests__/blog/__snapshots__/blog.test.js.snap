// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`storableFile 1`] = `
Object {
  "body": "<h1>Test Post</h1>

<p>the copy of the test post will go here</p>",
  "createdDate": 1485460800,
  "name": "Test Post Seo",
  "seo": "2017-01-27-test-post-seo",
  "updatedDate": 1482363367071,
}
`;
